### ISSUE TITLE
parseFromUrl doesn't exchange code for token(s)

### ISSUE THREAD

**Issue by Frozenlock** [okta/okta-auth-js#966]
From the docs:

> If an authorization code is present, **it will be exchanged for token(s)** by posting to the tokenUrl endpoint.

https://github.com/okta/okta-auth-js#tokenparsefromurloptions

However this isn't the case.
The promise returns `code`, `state`, but the `tokens` field is empty:

```js
{
    "tokens": {},
    "state": "EyxRSCIWRc7pPx7syhlMFzGhftFNyMQFw0j8Y2SjpW7PXQNp0YRPQ0oO6ZqKPX4g",
    "code": "20f3dc43-d1a6-4b2a-9f14-04c9c86b4c62"
}
```

Looking at the network activity, `Okta-auth-js` doesn't try to reach the `tokenUrl` endpoint.

**Comment by shuowu** [comment]
@Frozenlock Have you set `pkce: true` when calling the method?

**Comment by Frozenlock** [comment]
No, `pkce` is set to false when initializing `OktaAuth` and then the method is called without additional options.

**Comment by Frozenlock** [comment]
I must be using the wrong method then. The doc mentions that by setting `pkce` to false one can use the [autorization code flow](https://github.com/okta/okta-auth-js#authorization-code-flow-for-web-and-native-client-types).

What method should be used once the redirection is done and the code is in the query parameters?

**Comment by Frozenlock** [comment]
Also, I realize I was using the wrong method, but shouldn't it throw an exception in that case?

**Comment by lparisot** [comment]
Hello @Frozenlock, can you share what method must be used in this case ? I have the same problem ...

I use the https://github.com/okta/okta-auth-js/tree/master#authorization-code-flow-for-web-and-native-client-types flow with pkce set to false and handle the callback on a dedicated route https://github.com/okta/okta-auth-js/tree/master#handling-the-callback-with-path-routing-on-a-dedicated-route

**Comment by jaredperreault-okta** [comment]
@lparisot This is a somewhat old thread, do you mind creating a new issue and include as many relevant details as possible? For one, why you're choosing to not use `pkce`

**Comment by jaredperreault-okta** [comment]
@lparisot I am happy to assist you, but I need you to create a new issue
